### PR TITLE
Mg 69 collections

### PR DESCRIPTION
**Pull request by rastislav-chynoransky**
https://jira.sng.sk/browse/MG-69

### CODE REVIEW

File: models/Collection.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+export default class Collection extends BaseModel {
+  public declare id: string
+  public declare name: string
+  public declare text: string
+  public declare header_image_src: string
+  public declare header_image_srcset: string
+  public declare item_filter: string | string[]
+
+  items: Item[] = []
```
Comment (mayoforkovic) [inline models/Collection.ts:13]: Can you add zod schema parse? 
Example is in my last commits, Ill show you after standup.

File: plugins/collections.ts
Comment (mayoforkovic) [inline plugins/collections.ts:1]: This should be state managment for detail of collection and not index of collections. You have it reversed.

**Review by mayoforkovic — CHANGES_REQUESTED** [review]
Good direction but lets discus how to optimize collections plugin or maybe reuse controls from filter plugin we have already.

File: plugins/collections.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    total: computed(() => collectionsDataFetch.data.value?.meta.total ?? 0),
+    lastPage: computed(() => collectionsDataFetch.data.value?.meta.last_page ?? 0),
+  }
+}
+
+export default defineNuxtPlugin((nuxtApp) => {
+  nuxtApp.vueApp.provide(CollectionsSymbol, collectionsService())
+
+  return {}
+})
```
Comment (eronisko) [inline plugins/collections.ts:70]: Why is this a plugin and not just a plain composable?

**Review by eronisko — COMMENTED** [review]
I don't understand the need for plugin and injection (except it being consistent with previous code)

LGTM otherwise

**Review by mayoforkovic — APPROVED** [review]

**Pull request merged**